osmnx/osm_xml.py

```diff
@@ -195,6 +201,8 @@ def save_graph_xml(
 
     # rename columns per osm specification
     gdf_nodes.rename(columns={"x": "lon", "y": "lat"}, inplace=True)
+    gdf_nodes.lon = gdf_nodes.lon.round(coordinate_precision)
+    gdf_nodes.lat = gdf_nodes.lat.round(coordinate_precision)
```

Comment:
For Geo/DataFrame columns, please access them via `gdf["col_name"]` rather than `gdf.col_name`.